easybuild/tools/filetools.py

```diff
@@ -581,6 +645,15 @@ def download_file(filename, url, path, forced=False):
         timeout = 10
     _log.debug("Using timeout of %s seconds for initiating download" % timeout)
 
+    # parse option HTTP header fields for URLs containing a pattern
+    http_header_fields_urlpat = build_option('http_header_fields_urlpat')
+    # compile a dict full of {urlpat: [header, list]}
+    urlpat_headers = dict()
+    if http_header_fields_urlpat is not None:
+        # there may be multiple options given, parse them all, while updating urlpat_headers
+        for arg in http_header_fields_urlpat:
+            urlpat_headers = parse_http_header_fields_urlpat(arg, urlpat_headers)
```

Comment:
passing `urlpat_headers` as a positional argument, instead of keyword, will pass it to `urlpat` instead...

the reason this seems to work anyway is that the `urlpat_headers` default value is mutable, so `urlpat_headers` is persistent (e.g. https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html)

it would be better to define above

```python
def parse_http_header_fields_urlpat(arg, urlpat=None, header=None, urlpat_headers=None, maxdepth=3):
    ...
    if urlpat_headers is None:
        urlpat_headers = {}
```

Reply:
Thanks for your point.